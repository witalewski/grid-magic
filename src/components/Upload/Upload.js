import React, { Component } from 'react';
import { func } from 'prop-types';
import { observer, inject } from 'mobx-react';

class Upload extends Component {
  static propTypes = {
    addFile: func.isRequired,
  };

  render() {
    const { addFile } = this.props;
    return <input type="file" onChange={event => addFile(event.target.files[0])} />;
  }
}

export { Upload };
export default inject(({ appState }) => ({
  addFile: appState.addFile,
}))(observer(Upload));
