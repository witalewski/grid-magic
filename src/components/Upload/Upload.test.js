import React from 'react';
import { Upload } from './Upload';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Upload', () => {
  it('displays upload', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Upload addFile={jest.fn()} />
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
