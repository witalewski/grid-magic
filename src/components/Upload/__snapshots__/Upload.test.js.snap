// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload displays upload 1`] = `
<input
  className="file-upload"
  onChange={[Function]}
  type="file"
/>
`;
